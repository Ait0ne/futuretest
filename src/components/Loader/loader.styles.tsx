import styled, {keyframes} from 'styled-components';

export const SpinnerContainer = styled.div`
    width: 100vw;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: rgba(0,0,0,0.1);
    position: fixed;
    top: 0;
    left: 0;
    z-index:1;
`

const rotate = keyframes`
100% {
    transform: rotate(360deg);
}
`

const dash = keyframes`
  0% {
    stroke-dasharray: 1, 150;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 90, 150;
    stroke-dashoffset: -35;
  }
  100% {
    stroke-dasharray: 90, 150;
    stroke-dashoffset: -124;
  }
`


export const Spinner = styled.svg`
width:50px;
height:50px;
animation: ${rotate} 2s linear infinite;
>circle {
    stroke: rgba(0,0,255,0.5);
    stroke-linecap: round;
    animation: ${dash} 1.5s ease-in-out infinite;
}
`