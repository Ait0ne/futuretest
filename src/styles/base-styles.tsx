import styled from 'styled-components';

interface ButtonProps {
    active?: boolean
}

export const Button = styled.button<ButtonProps>`
cursor: pointer;
min-width: 64px;
padding: 6px 16px;
color: rgba(0, 0, 0, 0.87);
text-transform: uppercase;
font-weight: 500;
border-radius:4px;
border: none;
margin: 5px;
background-color: #e0e0e0;
box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 1px 5px 0px rgba(0,0,0,0.12);
&:disabled {
    color: rgba(0, 0, 0, 0.26)
}
`

export const Input = styled.input`
padding: 12px 25px 12px 14px;
border-radius: 4px;
width:25ch;
border: 1px solid rgba(0,0,0,0.3);
font-size: 1rem;
margin: 5px 10px;
&:focus {
    outline:none;
}

`