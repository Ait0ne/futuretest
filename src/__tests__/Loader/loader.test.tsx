import React from 'react';
import {shallow} from 'enzyme';
import Loader from '../../components/Loader/loader.component';
import {Spinner} from '../../components/Loader/loader.styles';


describe('loader', () => {
    const loader = shallow(<Loader/>)
    it('renders correctly', () => {
        expect(loader).toMatchSnapshot()
    })

    it('renders a spinner', ()=> {
        expect(loader.find(Spinner).length).toEqual(1)
    })

})
