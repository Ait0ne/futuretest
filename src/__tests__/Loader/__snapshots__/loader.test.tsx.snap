// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader renders correctly 1`] = `
<styled.div>
  <styled.svg
    viewBox="0 0 50 50"
  >
    <circle
      cx="25"
      cy="25"
      fill="none"
      r="20"
      strokeWidth="5"
    />
  </styled.svg>
</styled.div>
`;
